src/utils/getPartialCheckFromImportDiffs.ts

```diff
@@ -0,0 +1,89 @@
+import { reportWarning } from '../core/result';
+import NormalizedPath from '../types/NormalizedPath';
+import { FenceAndImportDiffs } from './getFenceAndImportDiffsFromGit';
+import * as path from 'path';
+import { PartialCheck } from '../types/PartialCheck';
+
+export function getPartialCheckFromImportDiffs(
+    graphDiff: FenceAndImportDiffs
+): PartialCheck | null {
+    let fences = new Set<NormalizedPath>();
+    let sourceFiles = new Set<NormalizedPath>();
+
+    let canResolve = true;
+
+    for (let [normalizedSourceFilePath, importDiff] of graphDiff.sourceImportDiffs.entries()) {
+        if (importDiff.addedImports?.length) {
+            // we need to re-check this file, since the new imports
+            // might violate the importing fence.
+            sourceFiles.add(normalizedSourceFilePath);
+        }
+    }
+
+    for (let [normalizedFencePath, fenceDiff] of graphDiff.fenceDiffs.entries()) {
+        if (fenceDiff.exports?.removed?.length) {
+            // if we removed an export, we have to re-evaluate all importers
+            // which mean we can't resolve from the repo diff
+            reportWarning(
+                `Cannot perform partial evaluation -- removed export(s) ${fenceDiff.exports.removed
+                    .map(x => {
+                        const v = { ...x };
+                        if (v.accessibleTo === null) {
+                            delete v.accessibleTo;
+                        }
+                        return JSON.stringify(v);
+                    })
+                    .join(', ')} from fence ${path.relative(process.cwd(), normalizedFencePath)}`
+            );
+            canResolve = false;
+        }
+
+        const fenceHadExportsSectionAdded =
+            fenceDiff.exports !== null &&
+            fenceDiff.exports.removed === null &&
+            fenceDiff.exports.added !== null;
+        if (fenceHadExportsSectionAdded) {
+            // if we added an exports section, we have to re-evaluate
+            // all importers, which means we can't resolve from the repo diff
+            reportWarning(
+                `Cannot perform partial evaluation -- added an exports section to fence ${normalizedFencePath}`
+            );
+            canResolve = false;
+        }
+
+        const fenceHadImportsSectionAdded =

```

Comment:
Same here, I don't think you can infer it this way.

Reply:
see above.